README.md

```diff
@@ -1,6 +1,166 @@
 
 # cleancall
 
-[![Codecov test coverage](https://codecov.io/gh/r-lib/cleancall/branch/master/graph/badge.svg)](https://codecov.io/gh/r-lib/cleancall?branch=master)
+> C Resource Cleanup via Exit Handlers
 
-Easy resource cleaning from C.
+<!-- badges: start -->
+
+![lifecycle](https://img.shields.io/badge/lifecycle-experimental-orange.svg)
+[![Travis build status](https://travis-ci.org/r-lib/cleancall.svg?branch=master)](https://travis-ci.org/r-lib/cleancall)
+[![Windows Build
+status](https://ci.appveyor.com/api/projects/status/p2jjoufya2e66oa5/branch/master?svg=true)](https://ci.appveyor.com/project/gaborcsardi/cleancall)
+[![](https://www.r-pkg.org/badges/version/cleancall)](https://cran.r-project.org/package=cleancall)
+[![CRAN RStudio mirror
+downloads](https://cranlogs.r-pkg.org/badges/cleancall)](https://www.r-pkg.org/pkg/cleancall)
+[![Coverage status](https://codecov.io/gh/r-lib/cleancall/branch/master/graph/badge.svg)](https://codecov.io/github/r-lib/cleancall?branch=master)
+<!-- badges: end -->
+
+## Features
+
+* Add exit handlers to a `.Call()` to C, via a `call_with_cleanup()`
+  wrapper.
+* Restrict an exit handler to run only on early exit (error, interrupt,
+  debugger exit, restart invokation, condition caught, etc.).
+* Exit handlers are executed in reverse order. Last added runs first.
+* Exit handlers can be added from any downstream function, they don't need
+  to be called directly from the function called by `call_with_cleanup()`.
+
+## Limitations
+
+We suggest that exit handlers are kept as simple and fast as possible.
+In particular, early exits _within_ exit handlers are not caught currently.
+If an exit handler exits early the others do not run. If this is an issue,
+you can wrap the exit handler in `R_ToplevelExec()`.
```

Comment:
But note that this solution is not ideal either because `R_ToplevelExec()` prevents R functions from handling the errors occurring from the handler.